Copy: Avoid extra dispatch in type-mismatch case

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69026
* #68963
* __->__ #68950
* #68949

Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

peterbell10 committed:
Copy: Avoid extra dispatch in type-mismatch case

[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"




[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"




[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"




[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"




[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064447).

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"


Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"


Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"


Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

[ghstack-poisoned]

peterbell10 committed:
Rebase and remove as_strided x-fail on "Copy: Avoid extra dispatch in type-mismatch case"


Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"


Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

[ghstack-poisoned]

peterbell10 committed:
Update on "Copy: Avoid extra dispatch in type-mismatch case"


Differential Revision: [D33064447](https://our.internmc.facebook.com/intern/diff/D33064447)

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064447).
